spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/graphite/GraphiteMetricsExportAutoConfiguration.java

```diff
@@ -58,9 +60,15 @@ public GraphiteConfig graphiteConfig(GraphiteProperties graphiteProperties) {
 
 	@Bean
 	@ConditionalOnMissingBean
-	public GraphiteMeterRegistry graphiteMeterRegistry(GraphiteConfig graphiteConfig,
-			Clock clock) {
-		return new GraphiteMeterRegistry(graphiteConfig, clock);
+	public HierarchicalNameMapper graphiteHierarchicalNameMapper(GraphiteConfig config) {
```

Comment:
This should declare that `GraphiteHierarchicalNameMapper` is returned to provide as much type information to the bean factory as possible.

Reply:
@wilkinsona I applied this requested change in fee2b05c9f85648b76eccf5fcfbb810cac7f83ba.